transform/operations_test.go

```diff
@@ -375,21 +386,122 @@ func TestTimeParse(t *testing.T) {
 	runOpTests(t, func() transformOperation { return &timeParse{} }, tests)
 }
 
+func TestCurrentTime(t *testing.T) {
+	tests := []opTests{
+		{
+			description: "Simple working case",
+			args:        map[string]string{"format": "RFC3339"},
+			want:        time.Now().Format(time.RFC3339),
+		},
+		{
+			description: "Missing arg",
+			args:        map[string]string{},
+			wantInitErr: true,
+		},
+		{
+			description: "Too many args",
+			args:        map[string]string{"format": "RFC3339", "cookies": "failure"},
+			wantInitErr: true,
+		},
+		{
+			description: "Format not predefined",
+			args:        map[string]string{"format": "Mon Jan 2 15:04:05 MST 2006"},
+			want:        time.Now().Format("Mon Jan 2 15:04:05 MST 2006"),
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.description, func(t *testing.T) {
+			op, err := runOpTestInit(func() transformOperation { return &currentTime{} }, test)
+			if err != nil {
+				return
+			}
+			got, err := runOpTestTransform(op, test)
+
+			if err != nil {
+				t.Fatal(err)
+				return
+			}
+
+			if test.want == nil {
+				return
+			}
+
+			wantResult, ok := test.want.(string)
+			if !ok {
+				t.Fatal("want must be a string")
+			}
+
+			gotResult, ok := got.(string)
+			if !ok {
+				t.Fatal("function must return string")
+			}
+
+			var gotParse time.Time
+			var wantParse time.Time
+
+			if test.args["format"] == "RFC3339" {
+				gotParse, err = time.Parse(time.RFC3339, gotResult)
+				wantParse, err = time.Parse(time.RFC3339, wantResult)
+				if err != nil {
+					t.Fatal(err)
+				}
+			} else {
+				gotParse, err = time.Parse(test.args["format"], gotResult)
+				wantParse, err = time.Parse(test.args["format"], wantResult)
+				if err != nil {
+					t.Fatal(err)
+				}
+			}
+			if result := compareTimeStamps(wantParse, gotParse); !result {
+				t.Fatal("time returned not close enough to current time")
+			}
+		})
+	}
+}
+
 func TestStringToInteger(t *testing.T) {
 	tests := []opTests{
 
 		{
 			description: "Simple working case",
-			in: "237754",
-			want: 237754,
+			in:          "237754",
+			want:        237754,
 		},
 
 		{
 			description: "Boolean input type",
-			in: true,
-			wantErr: true,
+			in:          true,
+			wantErr:     true,
 		},
-
 	}
 	runOpTests(t, func() transformOperation { return &stringToInteger{} }, tests)
 }
+
+func compareTimeStamps(time1 time.Time, time2 time.Time) bool {
+	maxTimeDifference := time.Duration(300) * time.Second
+	actualDiff := absValue(time1.Sub(time2))
```

Comment:
if you flip time2 and time 1 you shouldn't be absValue (which I'm pretty sure exists in math though it takes a float64)